crates/common/tedge_config/src/tedge_config_cli/settings.rs

```diff
@@ -260,6 +260,20 @@ impl ConfigSetting for MqttClientPortSetting {
     type Value = Port;
 }
 
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub struct MqttClientCafileSetting;
+
+impl ConfigSetting for MqttClientCafileSetting {
+    const KEY: &'static str = "mqtt.client.cafile";
+
+    const DESCRIPTION: &'static str = concat!(
+        "Path to the CA certificate used by MQTT clients to use when ",
```

Comment:
I would allow this path to be a directory of trusted certificates.

Reply:
I made it a path to a single certificate because when developing I had both CA certificate and server certificate in one directory, and didn't want to include both. But you're right that just 1 certificate might be too restrictive. Both MQTT clients that I used had both `cafile` and `capath` options, so perhaps I shall add `mqtt.client.capath` option?

Reply:
> However, I wonder if a single option can be used for both as the behavior can be adapted whether the path points to a file or a directory.

Having thought about it, I think with two separate options it is clear from the invocation whether or not we use a single certificate, or multiple certificates. With one option, you would have to look in the filesystem whether or not this path is a file or a directory, making the invocation ambiguous. For this reason I think it would be better to have two options.

Reply:
Wrt. the `capath` option, I have a question. How do we decide which certificates do we read and add? Do we attempt to parse every file in the directory as a certificate? Do we filter by file extensions, like `.pem` or `.crt`?

Here's what two example MQTT clients do:

- `mosquitto_pub`
  ```
  --capath
           Define the path to a directory containing PEM encoded CA certificates that are trusted. Used to enable SSL
           communication.

           For --capath to work correctly, the certificate files must have ".crt" as the file ending and you must run
           "openssl rehash <path to capath>" each time you add/remove a certificate.
  ```
- MQTT CLI
  It does not explain in the manpage, but it seems to [include all files with extensions `.pem`, `.cer`, and `.crt`](https://github.com/hivemq/mqtt-cli/blob/master/src/main/java/com/hivemq/cli/utils/CertificateConverterUtils.java#L32)

Reply:
> I think with two separate options it is clear from the invocation

By this, I meant "invocation" for CLI clients, and forgot that in the case of thin-edge, it is hidden inside configuration files. Still, having a distinction whether or not we use only one or multiple certificates, is what I would be in favour of keeping, even in a config file, unless there is some clear use case where one option that deals with both files and directories is superior.

Reply:
It makes senses to have these two options in the configuration file - as proposed by this PR.